src/actions/User.ts

```diff
@@ -56,3 +60,55 @@ export function chapterUpdateUser(userUpdate: ChapterUpdateUser) {
     },
   });
 }
+
+const MAX_UPLOAD_MB = 10;
+/**
+ * @param file - the file being uploaded
+ * @param nameWithoutExtension - the name to give the file in storage (without the extension). if two files have the same name
+ * for a user, the file will overwrite the existing file. if a name is not
+ * provided, a randomly generated name will be used that does NOT collide with an
+ * existing file
+ * @param onProgress - called on progress events. on fast networks, this
+ * will might just emit one event
+ *
+ * Will throw error if file is too large
+ */
+export async function uploadFile(
```

Comment:
I think it makes more sense to move this function to your uploaded-files.ts util file.  Let me know if you disagree though.

Reply:
Yep, that makes sense to me